flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkBase.java

```diff
@@ -234,4 +255,36 @@ private void checkErrorAndRethrow() {
 			throw new RuntimeException("An error occured in ElasticsearchSink.", cause);
 		}
 	}
+
+	private void retry(T value) throws Exception {
+		int retryCounter = 1;
+
+		while (retryCounter <= connectionRetries) {
+			if (bulkProcessor != null) {
+				bulkProcessor.close();
+				bulkProcessor = null;
+			}
+
+			if (client != null) {
+				client.close();
+			}
+
+			try {
+				open(null);
```

Comment:
This open() call seems a bit odd to me. I don't think its a good practice to call that here, since essentially its a life cycle method used by the system.